investos/portfolio/risk_model/factor_risk.py

```diff
@@ -1,23 +1,351 @@
+    "market_cap_open_dil_to_net_income_ltm": "7e4264a9-a137-4f5b-bc10-1f1e90d2f3f5",
+    "market_cap_open_dil_to_operating_excl_wc_cf_ltm": "5f050fce-5099-4ce9-a737-5d8f536c5826",  # ltm market_cap_open_dil to operating_excl_wc_cf multiple. ID is FactSet Sym ID.
+    # Growth
+    "sales_ltm_growth_over_sales_ltm_lag_1a": "e6035b0a-65b9-409d-a02e-b3d47ca422e2",
+}
 
 
 class FactorRisk(BaseRisk):
     """Multi-factor risk model."""
 
     def __init__(
-        self, factor_covariance, factor_loadings, idiosyncratic_variance, **kwargs
+        self,
+        start_date,
+        end_date,
+        risk_model_window: int,
+        recalc_freq: int,
+        fos_risk_factor_uuids: list = [],
+        fos_return_uuid: str = "return_1d",
+        adjustments: dict = {"FactorCovEstArgs": {"window": 21, "half_life": 360}},
+        factor_covariance=None,
+        factor_loadings=None,
+        idiosyncratic_variance=None,
+        **kwargs,
     ):
         super().__init__(**kwargs)
 
-        self.factor_covariance = factor_covariance
-        self.factor_loadings = factor_loadings
-        self.idiosyncratic_variance = idiosyncratic_variance
+        self._risk_model_window = risk_model_window
+        self._recalc_freq = recalc_freq
+
+        self._fos_risk_factor_uuids = fos_risk_factor_uuids
+        self._fos_return_uuid = fos_return_uuid
+
+        self._end_date = end_date
+        self._start_date = start_date
+        self._adjustments = adjustments
+
+        if self._fos_risk_factor_uuids:
+            print("Running Barra Risk model...")
+            self._pull_fos_risk_factor_dfs()
+            self._generate_risk_models()
+            self._apply_risk_model_adjustments()
+            print(
+                "\nDone generating factor loadings,factor covariance, and idiosyncratic variance"
+            )
+        else:
+            self._factor_covariance = factor_covariance
+            self._factor_loadings = factor_loadings
+            self._idiosyncratic_variance = idiosyncratic_variance
+
+    # Initialization
+    def _pull_fos_risk_factor_dfs(self):
+        # pull from feature hub
+        print("\n1. Pulling out data from FeatureHub...")
+
+        if not fos.api_key:
+            raise ValueError(
+                "Feature hub connection failed. Please check the API key is valid."
+            )
+
+        # Initialize a DataFrame
+        self._df_loadings = (
+            fos.Feature.get(fos_return_uuid_dict[self._fos_return_uuid])
+            .get_df()
+            .rename(columns={"value": self._fos_return_uuid})
+        )
+
+        for factor in self._fos_risk_factor_uuids:
+            temp_df = (
+                fos.Feature.get(fos_risk_factor_uuids_dict[factor])
+                .get_df()
+                .rename(columns={"value": factor})
+            )
+            if "datetime" not in temp_df.columns:
+                # Perform operations for non-datetime columns (assuming to create dummies)
```

Comment:
I think we can only assume dummies if no datetime and str or object value (i.e. is categorical). FeatureHub should label things if categorical as well. Let's take this logic out for now.

Reply:
We won't be using it anyway since we have Beta